app/models/pipeline_run.rb

```diff
@@ -1444,19 +1444,26 @@ def get_summary_contig_counts(min_contig_size)
   end
 
   def get_summary_contig_counts_v2(min_contig_size)
-    summary_dict = {} # key: count_type:taxid , value: contigs, contig_reads
-    contig_lineages(min_contig_size).each do |c|
-      lineage = JSON.parse(c.lineage_json)
-      lineage.each do |count_type, taxid_arr|
-        taxids = taxid_arr[0..1]
-        taxids.each do |taxid|
-          summary_per_tax = summary_dict[taxid] ||= {}
-          summary_per_tax_and_type = summary_per_tax[count_type.downcase] ||= {}
-          summary_per_tax_and_type[c.read_count] ||= 0
-          summary_per_tax_and_type[c.read_count] += 1
+    # key: taxid:count_type, value: read_counts: count
+    # Create and store default values for the hash if the key doesn't exist yet
+    summary_dict = Hash.new do |summary, taxid|
```

Comment:
Nit: Instead of naming them `taxid` and `tax_id`, can we do `tax_id` and `tax_id_hash`?
```
summary_dict = Hash.new do |summary_hash, tax_id|
  summary_hash[tax_id] = Hash.new do |tax_id_hash, count_type|
    tax_id_hash[count_type] = Hash.new do |count_type_hash, read_count|
```

Reply:
I think it makes it slightly more legible. This code block seems particularly confusing to read (with respect to the complexity of what it's actually doing), which is why I'm being nit-picky here.

Reply:
Done